lib/global-styles.php

```diff
@@ -840,6 +840,45 @@ function gutenberg_experimental_global_styles_register_cpt() {
 	register_post_type( 'wp_global_styles', $args );
 }
 
+/**
+ * Tell kses not to remove from the block markup (from style attribute)
+ * the CSS variables the block editor may add.
+ *
+ * @param array $allowed_attr List of allowed attributes.
+ * @return array Filtered result.
+ */
+function gutenberg_experimental_global_styles_allow_css_var_name( $allowed_attr ) {
+	return array_merge(
+		$allowed_attr,
+		array(
+			'--wp--style--color--link',
+		)
+	);
+}
+
+/**
+ * Tell kses to allow certain values for the properties
+ * the block editor may add.
+ *
+ * @param boolean $allow_css Whether or not this rule should be allowed.
+ * @param string  $css_test_string The CSS rule to process.
+ * @return boolean Filtered result.
+ */
+function gutenberg_experimental_global_styles_allow_css_var_value( $allow_css, $css_test_string ) {
+	$parts = explode( ':', $css_test_string, 2 );
+
+	if ( '--wp--style--color--link' !== $parts[0] ) {
+		return $allow_css;
+	}
+
+	// The only value the block editor attaches to link color is
+	// var(--wp--preset--color--<some-value-here>)
+	// so be specific in testing for that value.
+	return preg_match( '/var\(--wp--preset--color--.*\)/', $parts[1] );
```

Comment:
mmm, I believe we want to improve this bit. Allowing any characters (`.`) may still be problematic security-wise. It feels like we want to allow only alphanumeric and `-` as valid characters (eg of potential names: `...--some-color-value`, `...--some-color-value-2`, etc), which should cover all the potential color names that people use.

Reply:
improved at https://github.com/WordPress/gutenberg/pull/25411/commits/d27c8698c322aa007664861ac2ef37878104907b